modules/ensemble/lib/widget/collapsible.dart

```diff
@@ -0,0 +1,345 @@
+import 'package:ensemble/framework/error_handling.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/framework/view/data_scope_widget.dart';
+import 'package:ensemble/widget/helpers/controllers.dart';
+import 'package:flutter/material.dart';
+import 'package:accordion/accordion.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:yaml/yaml.dart';
+import 'package:ensemble/framework/widget/icon.dart' as ensembleIcon;
+
+class HeaderStyle {
+  Color? backgroundColor;
+  Color? backgroundColorOpened;
+  Color? borderColor;
+  Color? borderColorOpened;
+  double? borderWidth;
+  double? borderRadius;
+  EdgeInsets? padding;
+
+  HeaderStyle({
+    this.backgroundColor,
+    this.backgroundColorOpened,
+    this.borderColor,
+    this.borderColorOpened,
+    this.borderWidth,
+    this.borderRadius,
+    this.padding,
+  });
+}
+
+class ContentStyle {
+  Color? backgroundColor;
+  Color? borderColor;
+  double? borderWidth;
+  double? borderRadius;
+  double? horizontalPadding;
+  double? verticalPadding;
+
+  ContentStyle({
+    this.backgroundColor,
+    this.borderColor,
+    this.borderWidth,
+    this.borderRadius,
+    this.horizontalPadding,
+    this.verticalPadding,
+  });
+}
+
+class Collapsible extends StatefulWidget
+    with
+        Invokable,
+        HasController<CollapsibleController, CollapsibleState> {
+  static const type = 'Collapsible';
+
+  Collapsible({Key? key}) : super(key: key);
+
+  final CollapsibleController _controller = CollapsibleController();
+
+  @override
+  CollapsibleController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => CollapsibleState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'openSections': () => _controller.openSections,
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'items': (value) => _controller.items = List<YamlMap>.from(value),
+      'isAccordion': (value) => _controller.isAccordion = Utils.getBool(value, fallback: true),
+      'initialOpeningSequenceDelay': (value) =>
+          _controller.initialOpeningSequenceDelay = Utils.optionalInt(value),
+      'headerStyle': (value) => _controller.headerStyle = HeaderStyle(
```

Comment:
This won't work. If you set this dynamically it will not trigger binding because it is nested inside. You will need to use something similar to BoxAnimationComposite. That ensure any nested property set will trigger bindings.

But an easier way is to not use headerStyle nor contentStyle. Simply flatten them at the root and use something like headerBackGroundColor (for header) and backgroundColor (for content).

Reply:
@vusters yeah i understood, and fixed that using composite classes as wanted to keep the same structure as react and separate styles for both header and content, tested it, bindings are working now.